api/lib/getJSON.js

```diff
@@ -160,12 +161,33 @@ const getRaag = shabad => ({
   raagWithPage: shabad.RaagWithPage,
 });
 
-const getWriter = shabad => ({
-  writerId: shabad.WriterID,
-  gurmukhi: shabad.WriterGurmukhi,
-  unicode: shabad.WriterUnicode,
-  english: shabad.WriterEnglish,
-});
+const getWriter = (shabad, fullShabad) => {
```

Comment:
You've made a good fix that will work, but perhaps can be simplified. What if we just send `fullShabad`? We could then start with this like you have:
```js
const writerInfoRow = fullShabad.find(row => !!row.WriterID);
```
and then do one return with optional chaining and nullish coalescing (so as to not introduce a breaking change):
```js
return {
  writerId: writerInfoRow?.WriterID ?? null,
  gurmukhi: writerInfoRow?.WriterGurmukhi ?? null,
  unicode: writerInfoRow?.WriterUnicode ?? null,
  english: writerInfoRow?.WriterEnglish ?? null,
};
```
Thoughts?

Reply:
Agree it could be simplified. Made it explicit with the "short circuit" on the first row having the writer since we need to fix data too and didn't want to impact performance. Might be negligible performance wise regardless. 

optional chaining would require some transpiling since we run api on node version < 13. need to test newer version or convert to [typescript](https://github.com/KhalisFoundation/banidb-api/issues?q=is%3Aissue+is%3Aopen+type). 

I can simplify to make it more readable if that helps.

Reply:
Actually I like this way better for now sine changing `getWriter` signature would require refactoring in other places as well. Adding an optional param is more tolerant but I do need to do a `fullShabad` param check before calling find on it.